### PR TITLE
Created new LoggingInterceptor that doesn't log any errors related to…

### PR DESCRIPTION
**Pull request by hector2**
… SocketTimeoutException.

## Types of changes

- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] This is **NOT** translation related. (See [here](https://github.com/aminecmi/ReaderforSelfoss/pull/170#issuecomment-355715654))

This closes issue #281

### CODE REVIEW

File: app/src/main/java/apps/amine/bou/readerforselfoss/api/mercury/MercuryApi.kt [hunk 1]
```diff
@@ -12,11 +12,11 @@ class MercuryApi(shouldLog: Boolean) {
 
     init {
 
-        val interceptor = HttpLoggingInterceptor()
+        val interceptor = ApiLoggingInterceptor()
```
[the remark this answers is not in the document]
Comment (hector2) [reply app/src/main/java/apps/amine/bou/readerforselfoss/api/mercury/MercuryApi.kt]: Ok fixed

File: app/src/main/java/apps/amine/bou/readerforselfoss/api/selfoss/SelfossApi.kt [hunk 2]
```diff
@@ -95,12 +96,14 @@ class SelfossApi(
                 .setLenient()
                 .create()
 
-        val logging = HttpLoggingInterceptor()
+        HttpLoggingInterceptor()
```
[the remark this answers is not in the document]
Comment (hector2) [reply app/src/main/java/apps/amine/bou/readerforselfoss/api/selfoss/SelfossApi.kt]: Right, sorry T_T

File: app/src/main/java/apps/amine/bou/readerforselfoss/interceptors/ApiLoggingInterceptor.kt [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+                }
+            }
+        }
+
+        val startNs = System.nanoTime()
+        val response: Response
+        try {
+            response = chain.proceed(request)
+        } catch (e: SocketTimeoutException) {
+            throw e
```
[the remark this answers is not in the document]
Comment (hector2) [reply app/src/main/java/apps/amine/bou/readerforselfoss/interceptors/ApiLoggingInterceptor.kt]: New commit with a "simpler" way but I want your opinion

**Comment by hector2** [comment]
Hey @aminecmi I managed to solve this problem in a simpler but "hacky" way. What I do is, I use an interceptor that detects the timeout exception, then it forms an apparently good response with code 504 to bypass the logging interceptor, and finally there's another interceptor that throws a SocketTimeoutException if the status code is 504.

With this code, we don't log the timeout but still throw it in order to reach the retrofit callbacks (onFailure). If we don't throw any exception, we would have to check in every onResponse callback if is a timeout.

What do you think?

**Comment by aminecmi** [comment]
Hi @hector2 .

Thanks for the changes. This seems a lot more simple.

**Pull request merged**